stellargraph/core/graph.py

```diff
@@ -41,166 +41,232 @@
             edges, a DataFrame can be passed directly, and the type defaults to the
             ``edge_type_default`` parameter. Edges have an ID taken from the index of the dataframe,
-            and they have to be unique across all types. This must be used with the ``nodes``
-            argument. This uses the same basic structure as the ``node_features`` argument,
-            described above. **Warning**: this is experimental (see warning above for more details).
+            and they have to be unique across all types.
 
-        source_column: str, optional
+        is_directed (bool, optional):
+            If True, the data represents a directed multigraph, otherwise an undirected multigraph.
+
+        source_column (str, optional):
             The name of the column to use as the source node of edges in the ``edges`` edge list
             argument.
 
-        target_column: str, optional
+        target_column (str, optional):
             The name of the column to use as the target node of edges in the ``edges`` edge list
             argument.
 
-        edge_weight_label: str, optional
-            The name of the attribute to use as the weight of edges (for the `nodes`/`edges`
-            DataFrame parameters, this is the name of the column to use).
+        edge_weight_column (str, optional):
+            The name of the column in each of the ``edges`` DataFrames to use as the weight of
+            edges. If the column does not exist in any of them, it is defaulted to ``1``.
+
+        node_type_default (str, optional):
+            The default node type to use, if ``nodes`` is passed as a DataFrame (not a ``dict``).
+
+        edge_type_default (str, optional):
+            The default edge type to use, if ``edges`` is passed as a DataFrame (not a ``dict``).
+
+        dtype (numpy data-type, optional):
+            The numpy data-type to use for the features extracted from each of the ``nodes`` DataFrames.
+
+        graph:
+            Deprecated, use :meth:`from_networkx`.
+        node_type_name:
+            Deprecated, use :meth:`from_networkx`.
+        edge_type_name:
+            Deprecated, use :meth:`from_networkx`.
+        feature_name:
+            Deprecated, use :meth:`from_networkx`.
+        target_name:
+            Deprecated, use :meth:`from_networkx`.
+        edge_weight_label:
+            Deprecated, use :meth:`from_networkx`.
+        node_features:
+            Deprecated, use :meth:`from_networkx`.
     """
 
     def __init__(
         self,
-        graph=None,
+        nodes=None,
+        edges=None,
+        *,
         is_directed=False,
-        edge_weight_label="weight",
-        node_type_name=globalvar.TYPE_ATTR_NAME,
-        edge_type_name=globalvar.TYPE_ATTR_NAME,
+        source_column=globalvar.SOURCE,
+        target_column=globalvar.TARGET,
+        edge_weight_column=globalvar.WEIGHT,
```

Comment:
I've named this argument `edge_weight_column` rather than `weight_column` so that it's not ambiguous about being a "weight" for nodes  (e.g. a class weight for machine learning, or a scaling parameter for node features or something), however this means it's somewhat inconsistent with `source_column` and `target_column`.

I see our options as:

1. leave it like this (`source_column`, `target_column`, `edge_weight_column`)
2. put `edge` on everything (`edge_source_column`, `edge_target_column`, `edge_weight_column`)
3. leave `edge` off everything (`source_column`, `target_column`, `weight_column`)
4. some other names that are magically not ambiguous?!

Any thoughts this?

Also, is it worth being consistent between this and `from_networkx`? After this change, `from_networkx`'s equivalent argument is different (`edge_weight_label` here, and after #893, `edge_weight_attr`). Does that matter?

Reply:
Option 1 doesn't occur to me as being particularly inconsistent - looks fine IMO!

As for the consistency between this and networkx, I see the parameters as referring to some property in the "input" rather than the result, so I think `column` makes sense here since we're referring to a column in the input dataframe, whereas in networkx it's an attribute in the input networkx graph.

Reply:
I think option 1. `edge_weight_column` is good, since a node could have a weight, but not a source or target.

Reply:
Ok, I'll leave it as is (option 1). 👍